spyder/plugins/completion/plugin.py

```diff
@@ -156,19 +157,35 @@ def send():
                     or [-1])
                 send = req_id == max_req_id
 
+            logger.debug("Completion plugin: Request {} "
+                         "removed".format(req_id))
             del self.requests[req_id]
 
             # Response only to recent requests.
             if send:
                 self.gather_and_send(request_responses)
 
+        # Skip the waiting logic below when fallback is the only
+        # client that works for language
+        if self._is_fallback_only(language):
```

Comment:
this seems like a good approach

Reply:
Thanks! It took me several hours of hard thinking to come up with it.